src/Microsoft.Azure.SignalRService/Credential.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.SignalRService
+{
+    public class Credential
+    {
+        public string Endpoint { get; set; }
+        public string AccessKey { get; set; }
+
+        private static readonly char[] PropertySeparator = { ';' };
+        private static readonly char[] KeyValueSeparator = { '=' };
+        private const string EndpointProperty = "endpoint";
+        private const string AccessKeyProperty = "accesskey";
+
+        public Credential()
+        {
+        }
+
+        internal void ParseConnectionString(string connectionString)
+        {
+            var properties = connectionString.Split(PropertySeparator, StringSplitOptions.RemoveEmptyEntries);
+            if (properties.Length > 1)
+            {
+                var dict = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+                foreach (var property in properties)
+                {
+                    var kvp = property.Split(KeyValueSeparator, 2);
+                    if (kvp.Length != 2) continue;
+
+                    var key = kvp[0].Trim();
+                    if (dict.ContainsKey(key))
+                    {
+                        throw new ArgumentException($"Duplicate properties found in connection string: {key}.");
+                    }
+
+                    dict.Add(key, kvp[1].Trim());
+                }
+
+                if (dict.ContainsKey(EndpointProperty) && dict.ContainsKey(AccessKeyProperty))
+                {
+                    Endpoint = dict[EndpointProperty].TrimEnd('/');
+                    AccessKey = dict[AccessKeyProperty];
+                }
+            }
+
+            throw new ArgumentException($"Connection string missing required properties {EndpointProperty} and {AccessKeyProperty}.");
```

Comment:
I don't see return in the code above, so this will always be executed?

Reply:
Reuse `ConnectionStringParser` now